### PR TITLE
ci(jenkins): run tests without async hooks on Node.js nightly

### PR DESCRIPTION
**Pull request by watson**
The diff might be a little hard to read, but I essentially just duplicated the "Nightly Test" stage.

I've also split up the edge yaml file so that we in the future can test different Node.js versions for RC and Nightly (which we will have to the moment this PR lands - just felt it was cleaner to do in two PRs)

### CODE REVIEW

**Comment by watson** [comment]
@v1v is there a way we can manually trigger the nightly + rc builds on Jenkins on this PR so that we can see it works?

**Comment by watson** [comment]
I assume the current [`continuous-integration/jenkins/pr-merge`](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/PR-1277/1/pipeline/675) failure is just something to do with instability in our Jenkins setup?

**Comment by watson** [comment]
jenkins run the tests please

File: Jenkinsfile [hunk 1]
```diff
@@ -173,7 +173,29 @@ pipeline {
...
+                  parallel(parallelTasks)
+                }
+              }
+            }
+          }
+        }
+        stage('Nightly Test - No async hooks') {
+          agent { label 'docker && immutable' }
+          environment {
+            NVM_NODEJS_ORG_MIRROR = "https://nodejs.org/download/nightly/"
```
Comment (v1v) [inline Jenkinsfile]:
```suggestion
            NVM_NODEJS_ORG_MIRROR = "https://nodejs.org/download/nightly/"
            ELASTIC_APM_ASYNC_HOOKS = "false"
```

File: Jenkinsfile [hunk 2]
```diff
@@ -213,12 +235,12 @@ pipeline {
...
+            withGithubNotify(context: 'RC No Async Hooks Test', tab: 'tests') {
               deleteDir()
               unstash 'source'
               dir("${BASE_DIR}"){
                 script {
-                  def node = readYaml(file: '.ci/.jenkins_edge_nodejs.yml')
+                  def node = readYaml(file: '.ci/.jenkins_rc_nodejs.yml')
                   def parallelTasks = [:]
                   node['NODEJS_VERSION'].findAll{ it != '6' }.each{ version ->
                     parallelTasks["Node.js-${version}-nightly-no_async_hooks"] = generateStep(version: version, edge: true)
```
Comment (v1v) [inline Jenkinsfile]:
```suggestion
                    parallelTasks["Node.js-${version}-rc-no_async_hooks"] = generateStep(version: version, edge: true)
```

**Comment by v1v** [comment]
> @v1v is there a way we can manually trigger the nightly + rc builds on Jenkins on this PR so that we can see it works?

@watson , yes, it's possible to manually build this PR with all the RC/Nightly stages as long as the below two params are selected when using the Jenkins UI, parameters:
- `Run_As_Master_Branch
- `test_edge_ci `

For instance, I just triggered manually a build for this PR with the below parameters:
- [build #3](https://apm-ci.elastic.co/job/apm-agent-nodejs/job/apm-agent-nodejs-mbp/job/PR-1277/3/parameters/)

I've decided to only validate the RC/Nightly without running any other tests (aka neither the test stage nor TAV stage)

**Comment by watson** [comment]
@v1v Inspired by your review I just made a few other changes to the `parallelTasks` names. I think they are ok?

**Comment by watson** [comment]
@v1v I somehow can't get the new "no async hooks" group to run without Node.js 6. I tried to copy the logic from the other similar test group, but [when I run](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/PR-1277/9/pipeline) it, it still includes Node.js 6. Can you help me figure out why?

**Comment by watson** [comment]
@v1v If I read this screenshot correctly, it looks like the 4 different edge-test groups doesn't run in parallel, but sequentially. Do that be easy to fi in this PR, or do you suggest we fix that in another PR instead?

image

**Comment by v1v** [comment]
> ..  Do that be easy to fi in this PR?

@watson , it could be potentially included in this PR with something like the below snippet:

```
    stage('Edge Test') {
      agent none
      options { skipDefaultCheckout() }
      environment {
        HOME = "${env.WORKSPACE}"
      }
      when {
        beforeAgent true
        allOf {
          anyOf {
            expression { return params.Run_As_Master_Branch }
            triggeredBy 'TimerTrigger'
          }
          expression { return params.test_edge_ci }
        }
      }
      parallel {
        stage('Nightly Test') {
           ... 
        }
        stage('Nightly Test - No async hooks') {
           ... 
        } 
        stage('RC Test') {
           ... 
        } 
        stage('RC Test - No async hooks') {
           ... 
        } 
      }
   }
```

Although I have not tested it out, we use the parallel within the stage step in some other pipelines, so I guess it should work out of the box, I'm happy to modify this PR to include this particular use case if you want

**Comment by watson** [comment]
@v1v To make the diff simpler, I think it's better to make it as a separate PR. Let's just get this one merged first if I could get your final review :)

**Pull request merged**